### PR TITLE
deps: bump go-libp2p and all supporting pkgs

### LINKED ISSUE #1029: deps: update go-libp2p to v0.24.0
There is a new release that merges go-libp2p-core into go-libp2p. We rely on go-libp2p-core in multiple places, so updating will require multiple reimports and some efforts. All the related libp2p deps should be updated as well, e.g pubsub

https://github.com/libp2p/go-libp2p/releases/tag/v0.22.0

### PR DESCRIPTION
**Pull request by Wondertan**
Based on #1546 

Closes #1029 

- [x] Enable quicv1

### CODE REVIEW

File: nodebuilder/p2p/misc.go [hunk 1]
```diff
@@ -4,9 +4,9 @@ import (
 	"time"
 
 	"github.com/ipfs/go-datastore"
-	coreconnmgr "github.com/libp2p/go-libp2p-core/connmgr"
-	"github.com/libp2p/go-libp2p-core/peerstore"
-	"github.com/libp2p/go-libp2p-peerstore/pstoremem"
+	connmgri "github.com/libp2p/go-libp2p/core/connmgr"
```
Comment (walldiss) [inline nodebuilder/p2p/misc.go:7]: `connmgri` - last letter is a typo?

Comment (Wondertan) [reply nodebuilder/p2p/misc.go:7]: stands for interface

**Review by walldiss — APPROVED** [review]

**Review by vgonkivs — DISMISSED** [review]

**Review by vgonkivs — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
Woah - i'm surprised nothing had to change!

**Comment by Wondertan** [comment]
Because I extracted everything out :)

**Pull request merged**